tests-beta/ios/MASVS-STORAGE/MASTG-TEST-0x55-2.md

```diff
@@ -0,0 +1,25 @@
+---
+platform: ios
+title: Sensitive Data Present In Keyboard Cache
+id: MASTG-TEST-0x55-2
+type: [dynamic]
+weakness: MASWE-0053
+---
+
+## Overview
+
+This test checks whether the keyboard cache contains sensitive data from the target app.
+
+## Steps
+
+1. Reset your iOS device keyboard cache by navigating to `Settings > General > Reset > Reset Keyboard Dictionary`.
+2. Exercise the application and identify the functionalities that allow users to enter sensitive data.
+3. Use @MASTG-TECH-0052 to retrieve the keyboard cache file with the extension `.dat` at `/private/var/mobile/Library/Keyboard/` and its subdirectories.
```

Comment:
was this working? which iOS version was it?

Reply:
Yes, it worked on iOS 15.4.1 iPhone XS. The file is `/private/var/mobile/Library/Keyboard/en-dynamic.lm`.
It's not very consistent though. I can't find there all suggestions that my keyboard cached(yes I restarted my device).